grails-app/controllers/org/pih/warehouse/core/StockMovementItemsParamsCommand.groovy

```diff
@@ -0,0 +1,19 @@
+package org.pih.warehouse.core
+
+import grails.validation.Validateable
+import org.pih.warehouse.api.PaginationCommand
+
+class StockMovementItemsParamsCommand extends PaginationCommand implements Validateable {
+
+    String id
+
+    Integer stepNumber
+
+    Boolean refresh = true
+
+    static constraints = {
+        stepNumber nullable: true
+        refresh nullable: true
+    }
```

Comment:
those two commands seem to be almost the same (the above could extend this one)

Reply:
I noticed that and also thought about extending `StockMovementItemsParamsCommand` from `StockMovementItemParamsCommand` but I didn't want two commands to depend on each other that much.
I'd rather stick with two separate DTOs without extending them from each other since both of them call different service methods.

Reply:
Yeah this is one of the reasons I don't like Command/DTOs as we can easily get into a situation where we create multiple Commands/DTOs for similar things. Until we can come up with a proper convention, I'd prefer have the controller action handle the request parameters itself in order to avoid the Indistinguishable Command Class per Controller Action :tm: anti-pattern 

> [!IMPORTANT]
> Convention Proposal:
> We shouldn't need to create a separate command class for every action. We should either be able to consolidate command classes for similar actions or deal with request parameters within the controller. 

But instead of continuing to litigate that here, let's move the discussion to the Naming Convention > Command Class document I created a few days ago.
https://pihemr.atlassian.net/wiki/spaces/OB/pages/3075997703/Command+Class

Once we have a satisfactory convention we can go back and apply that to our command class hierarchy (this is also very much related to the API resource refactoring that needs to happen as that will likely lead to a bunch of new command/DTOs as well).